### PR TITLE
Reorder C# script properties to fix editor serialization

### LINKED ISSUE #96983: Reordering properties from #93779 unsets editor set custom properties for [Tool] nodes.
### Tested versions

- Reproducible in 4.4 after #93779

### System information

Godot v4.4.dev.mono (40d8d82b8) - Ubuntu 24.04.1 LTS 24.04 - Wayland - Vulkan (Forward+) - integrated AMD Radeon Graphics (RADV RENOIR) - AMD Ryzen 7 5800H with Radeon Graphics (16 Threads)

### Issue description

The reordering done by #93779 seems to unset the custom properties set via the editor for `[Tool]` annotated scripts.

### Steps to reproduce

Create some node, attach a C# script with some custom property and annotate it with `[Tool]`.

After building, set the property in the editor. When reloading the scene, it will be unset/set to the default value. 

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/user-attachments/files/17002599/mrp.zip)

### PR DESCRIPTION
**Pull request by hayahane**
Fixes #96983

### CODE REVIEW

**Review by raulsntos — COMMENTED** [review]
I tested these changes and they fix the linked issue. However, properties exported by overriding `_get_property_list` now appear under the base class properties:

| 4.3.0 | This PR |
|-|-|
|  |  |

But this was also introduced by https://github.com/godotengine/godot/pull/93779 and we missed it.

**Comment by hayahane** [comment]
> I tested these changes and they fix the linked issue. However, properties exported by overriding `_get_property_list` now appear under the base class properties:

Pushed a fix for this. And I also found when `_get_property_list` is overridden, the base class's `_get_property_list` doesn't work. However in gdscript it works. Here is a similar test with derived class in #70146.
I only enable it for the derived class since I don't know how to call the base class's `_get_property_list`.
| GDScript | C# |
|----|-----|
|image| image |

**Comment by raulsntos** [comment]
> And I also found when `_get_property_list` is overridden, the base class's `_get_property_list` doesn't work.

That's because in Godot `_get_property_list` is a multi-level function and it's treated as such in GDScript, but C# doesn't treat it as a multi-level function and you have to call `base._GetPropertyList` from the implementation.

But it would be a more involved fix for a pre-existing issue (https://github.com/godotengine/godot/issues/75271), so it can handled in a follow-up PR.

**Comment by hayahane** [comment]
In this pr, only the `_get_property_list` overridden in the script will be called, and those from base classes will not. This should fix previous property ordering issue leaving #75271 for a future pr. (Currently don't have any idea how to call a method in base class from cpp side.)
Does this behaviour fit?

File: modules/mono/csharp_script.cpp [hunk 1]
```diff
@@ -30,6 +30,8 @@
 
 #include "csharp_script.h"
 
+#include "core/error/error_macros.h"
+#include "core/object/object.h"
```
Comment (raulsntos) [inline modules/mono/csharp_script.cpp:34]: These are unused.

```suggestion
```

**Review by raulsntos — COMMENTED** [review]
Yes, I think this is an improvement over what we had before. Thanks!

**Review by raulsntos — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**